packages/oid4vci-demo-frontend/src/components/InputField/index.tsx

```diff
@@ -0,0 +1,68 @@
+import React, {ChangeEvent, FC, HTMLInputTypeAttribute, ReactElement, useState} from 'react';
+import style from './index.module.css';
+import {FormFieldValue} from '../../types';
+
+type Props = {
+    type: HTMLInputTypeAttribute;
+    label?: string
+    defaultValue?: FormFieldValue
+    placeholder?: string
+    readonly?: boolean
+    customValidation?: RegExp
+    onChange?: (value: FormFieldValue) => Promise<void>
+}
+
+const InputField: FC<Props> = (props: Props): ReactElement => {
+    const {
+        readonly = false,
+        defaultValue,
+        placeholder,
+        label,
+        onChange,
+        type,
+        customValidation
+    } = props
+    const [value, setValue] = useState<FormFieldValue>(defaultValue)
+    const [isValid, setIsValid] = useState<boolean>(true)
+    const isCheckBox = type === 'checkbox'
+
+    const onChangeValue = async (event: ChangeEvent<HTMLInputElement>): Promise<void> => {
+        const value: string | boolean = type === 'checkbox' ? event.target.checked : event.target.value
+        setValue(value)
+
+        if (onChange) {
+            await onChange(value)
+        }
+    }
+
+    // FIXME fix the type complaining about FocusEvent<HTMLInputElement, Element>
```

Comment:
Does this work?
```
 const onBlur = async (event: React.FocusEvent<HTMLInputElement>): Promise<void> 
 ```

Reply:
no, i tried that but then it also complains:  Type 'FocusEvent' is not generic.

Reply:
ok this is weird. atleast to me. so importing FocusEvent and using: FocusEvent<HTMLInputElement>, does not work. but React.FocusEvent<HTMLInputElement> does work....

Reply:
ok, it was just not importing because i think there is a DOM variant. adding the import from react fixes it.